src/DockContainerWidget.h

```diff
@@ -214,6 +206,14 @@ class ADS_EXPORT CDockContainerWidget : public QFrame
 	 */
 	QList<CDockAreaWidget*> openedDockAreas() const;
 
+    /**
+     * This function returns true if this dock area has only one single
+     * visible dock widget.
+     * A top level widget is a real floating widget. Only the isFloating()
+     * function of top level widgets may returns true.
+     */
+    bool hasTopLevelDockWidget() const;
+

```

Comment:
I had to make this function public to be able to disable the "Detach" menu option when there is only one tab in a floating dock area.